server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -64,6 +64,7 @@
 
 public class EqualityExtractor {
 
+    private static final int MAX_ITERATIONS = 100_000;
```

Comment:
How long does it typically take to reach the 100_000 iterations?

Reply:
> 100k iterations for the query mentioned on the issue: https://github.com/crate/crate/issues/16066
takes a bit less than 3secs on my machine and all DocKeys are extracted.

I'd aim something closer to ~50-300ms as max before aborting the optimization

Reply:
Sorry, I posted the comment: https://github.com/crate/crate/pull/16098#issuecomment-2147563981 at the same time you commented :)

Reply:
> > 100k iterations for the query mentioned on the issue: #16066
> > takes a bit less than 3secs on my machine and all DocKeys are extracted.
> 
> I'd aim something closer to ~50-300ms as max before aborting the optimization

Hm ok, I will do some more tests and lower it accordingly, thx!

Reply:
I'm going with 10k, takes between 250 and 280 millis locally.